keps/sig-instrumentation/2305-metrics-cardinality-enforcement/README.md

```diff
@@ -236,19 +264,145 @@ This would then be interpreted by our machinery as this:
+  are missing a bunch of machinery and tooling and can't do that now.
+  No.
+* **Is the rollout accompanied by any deprecations and/or removals of features, APIs, 
+fields of API types, flags, etc.?**
+  A component metric flag for ingesting allowlist to be added.
+### Monitoring Requirements
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How can an operator determine if the feature is in use by workloads?**
+  The out-of-bound data will be recorded with label "unexpected" rather than the specific value.
+* **What are the SLIs (Service Level Indicators) an operator can use to determine 
+the health of the service?**
+  None.
+
+* **What are the reasonable SLOs (Service Level Objectives) for the above SLIs?**
+  None.
+
+* **Are there any missing metrics that would be useful to have to improve observability 
+of this feature?**
+  None.
+
+### Dependencies
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **Does this feature depend on any specific services running in the cluster?**
+  No.
+
+
+### Scalability
+
+_For alpha, this section is encouraged: reviewers should consider these questions
+and attempt to answer them._
+
+_For beta, this section is required: reviewers must answer these questions._
+
+_For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field._
+
+* **Will enabling / using this feature result in any new API calls?**
+  No.
+
+* **Will enabling / using this feature result in introducing new API types?**
+  Describe them, providing:
+  No.
+
+* **Will enabling / using this feature result in any new calls to the cloud 
+provider?**
+  No.
+* **Will enabling / using this feature result in increasing size or count of 
+the existing API objects?**
+  No.
+
+* **Will enabling / using this feature result in increasing time taken by any 
+operations covered by [existing SLIs/SLOs]?**
+  No.
+
+* **Will enabling / using this feature result in non-negligible increase of 
+resource usage (CPU, RAM, disk, IO, ...) in any components?**
+  No.
+
+### Troubleshooting
+
+The Troubleshooting section currently serves the `Playbook` role. We may consider
+splitting it into a dedicated `Playbook` document (potentially with some monitoring
+details). For now, we leave it here.
 
-## Graduation Criteria
+_This section must be completed when targeting beta graduation to a release._
 
-todo
+* **How does this feature react if the API server and/or etcd is unavailable?**
+  No additional impact comparing to what already exists.
+* **What are other known failure modes?**
+  None.
 
+* **What steps should be taken if SLOs are not being met to determine the problem?**
+  None.
 
-## Post-Beta tasks
+## Implementation History
 
-todo
+2020-04-15: KEP opened
 
-## Implementation History
+2020-05-19: KEP marked implementable
 
-todo
+v1.20: Launched in Alpha
```

Comment:
This is not yet launched; 1.21 is the first release we're targeting for implementation.

Reply:
removed.